sycl/include/CL/sycl/detail/defines.hpp

```diff
@@ -31,3 +37,25 @@
 #ifndef SYCL_EXTERNAL
 #define SYCL_EXTERNAL
 #endif
+
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__) && __has_builtin(__builtin_assume)
+#define __SYCL_ASSUME_INT(x) __builtin_assume(x <= INT_MAX)
+#define __SYCL_ASSUME_ID_INT(dims, x)                                          \
+  do {                                                                         \
+    for (size_t dim = 0; dim < dims; ++dim) {                                  \
+      size_t id = x.get_id(dim);                                               \
+      __SYCL_ASSUME_INT(id);                                                   \
+    }                                                                          \
+  } while (0)
+#define __SYCL_ASSUME_ARRAY_INT(dims, x)                                       \
+  do {                                                                         \
```

Comment:
why do we need do while(0); block here?

Reply:
As you may see there is no semicolon after `while (0)` which explicitly makes the user to put semicolon at the end of statement.

Reply:
The block is still executed due to post-check. Though the block is executed only once.